File: source/zombie/iso/objects/interfaces/Thumpable.java
// Decompiled on Sat Jan 17 08:24:00 EST 2026 with Zomboid Decompiler v0.2.3 using Vineflower.
package zombie.iso.objects.interfaces;

import zombie.characters.IsoGameCharacter;
import zombie.inventory.types.HandWeapon;
import zombie.iso.IsoMovingObject;

public interface Thumpable {
    boolean isDestroyed();

    void Thump(IsoMovingObject thumper);

    void WeaponHit(IsoGameCharacter chr, HandWeapon weapon);

    Thumpable getThumpableFor(IsoGameCharacter chr);

    float getThumpCondition();
}

File: converted/src/zombie/iso/objects/interfaces/Thumpable.cpp
#include "zombie/iso/objects/interfaces/Thumpable.h"

namespace zombie {
namespace iso {
namespace objects {
namespace interfaces {

bool Thumpable::isDestroyed() {
 // TODO: Implement isDestroyed
 return false;
}

void Thumpable::Thump(IsoMovingObject thumper) {
 // TODO: Implement Thump
}

void Thumpable::WeaponHit(IsoGameCharacter chr, HandWeapon weapon) {
 // TODO: Implement WeaponHit
}

Thumpable Thumpable::getThumpableFor(IsoGameCharacter chr) {
 // TODO: Implement getThumpableFor
 return nullptr;
}

float Thumpable::getThumpCondition() {
 // TODO: Implement getThumpCondition
 return 0;
}

} // namespace interfaces
} // namespace objects
} // namespace iso
} // namespace zombie
